bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ChannelDefinition.java

```diff
@@ -56,16 +58,13 @@ public ChannelDefinition(String id, ChannelTypeUID channelTypeUID) throws Illega
      *
      * @throws IllegalArgumentException if the ID is null or empty, or the type is null
      */
-    public ChannelDefinition(String id, ChannelTypeUID channelTypeUID, Map<String, String> properties, String label,
-            String description) throws IllegalArgumentException {
-        if ((id == null) || (id.isEmpty())) {
+    public ChannelDefinition(@NonNull String id, @NonNull ChannelTypeUID channelTypeUID,
+            @Nullable Map<String, String> properties, @Nullable String label, @Nullable String description)
```

Comment:
Didn't we agree on that we only want to state if a parameter is `NonNull` and if if can be `null` we omit the `@Nullable` annotation?